chapters/mime-type/index.rst

```diff
@@ -1,97 +1,118 @@
-    # assigning an invalid mime type leads to a ``ValueError``
-    d.mime_type = 'invalid/type' # raises exception
-
-Drivers derived from ``SegmentDriver`` can access the meme type of the documents to treat them differently.
-
-.. highlight:: python
-.. code-block:: python
-    from ..types.document import Document
-    from .. import DocumentSet
-
-    class SpecialSegmentDriver(SegmentDriver):
-        def __init__(
-                self,
-                *args,
-                **kwargs
-        ):
-            super().__init__(*args, **kwargs)
-
-        def _apply_all(self, leaves, *args, **kwargs):
-            docs = DocumentSet.flatten(leaves)
-            for doc in docs:
-                if doc.mime_type == 'text/plain':
-                    _args_dict = doc.get_attrs(*self.exec.required_keys)
-                    ret = self.exec_fn(**_args_dict)
-                    if ret:
-                        self._update(doc, ret)
+.. confval:: auto_assignment.py
+
+    .. highlight:: python
+    .. code-block:: python
+
+        from jina import Document
+
+        d1 = Document()
+        d1.text = 'my text 📩'
+        print(d1.mime_type)  # mime_type is 'text/plain'
+
+        d2 = Document()
+        d2.uri = 'https://upload.wikimedia.org/wikipedia/commons/a/a9/Example.jpg'
+        print(d2.mime_type)  # mime_type is 'image/jpeg'
+
+        d3 = Document()
+        d3.buffer = (b'\x89PNG\r\n\x1a\n\x00\x00\x00\rIHDR...')
+        print(d3.mime_type)  # mime_type is 'image/png'
+
+Manual mime type assignment
+---------------------------
+To have the full control over the mime type, it is also possible to set it manually like shown in the following example:
+
+.. confval:: manual_assignment.py
+
+    .. highlight:: python
+    .. code-block:: python
+
+        from jina import Document
 
+        svg_content = (
+            '<svg height="100" width="100">'
+            '  <circle cx="50" cy="50" r="40" stroke="black" stroke-width="3" fill="red" />'
+            '</svg>'
+        ).encode('utf8')
+        d = Document()
+        d.buffer = (svg_content)
+        print(d.mime_type)  # automatically detects that mime_type is 'image/svg'
+        # the mime type can be overwritten
+        d.mime_type = 'text/plain'
+        # assigning an invalid mime type leads to a ``ValueError``
+        d.mime_type = 'invalid/type' # raises exception
+
+Mime type in chunks
+-------------------
 For a variety of use cases, chunks have the same mime type as their parent documents.
 Examples are text documents where each sentence (chunk) is a text document on its own.
 Also when segmenting images or audio, most applications are creating chunks of the same mime type.
 However, there are scenarios where it makes sense to change the mime type on chunk level.
-A common example would be video documents where the chunks are images.
+Common examples would be for instance video documents where the chunks are images or images with text chunks (OCR).
```

Comment:
It is not so clear if OCR would need to be a Segmenter or a Crafter inmho, so it could be missleading?

Reply:
Yes. You are right. We first have to see how we would implement OCR.

Reply:
same with video to image segmentation, It is also doing segmenting and crafting in one run. The video has to be decoded (crafting), split into images (segmenting) and encoded into images (crafting).
I will bring up a discussion on how to design the interface of the crafter and segmenter.

Reply:
I removed the ocr example